### ISSUE TITLE
SIRIUS export

### ISSUE THREAD

**Issue by tomas-pluskal** [mzmine/mzmine2#371]
The SIRIUS export module does not report progress correctly. 
Also, it dies with a NullPointerException in case some scans don't have the mass list of given name.

<img width="1000" alt="screen shot 2017-10-07 at 13 51 15" src="https://user-images.githubusercontent.com/1796692/31310468-bf0e5172-ab66-11e7-87af-763657d9a603.png">

**Comment by tomas-pluskal** [comment]
@lfnothias @kaibioinfo

**Comment by tomas-pluskal** [comment]
I also tried running it on one of my datasets (~7000 detected peaks). It produced this MGF file (~5 GB): https://www.dropbox.com/s/qmla8zrtvdzawlv/sirius.mgf?dl=0
However, loading the file into SIRIUS does not seem to distinguish the different peaks - it loads up as a single compound.

**Comment by lfnothias** [comment]
Yeah that is right. We knew that it would be problematic with many samples/features. 

However, so far it works fine with Sirius on my end with very large files (I have not tried your file). Did you try Sirius CLI ou GUI ?

The module is exporting all MS1 and MS2 scans associated to each features, so it can be quite massive and redundant. Kai mentioned he intented to zip the .mgf to decrease the sample size.

**Comment by lfnothias** [comment]
@kaibioinfo To fix that issue, could we set a user-defined value for the number of MS1/MS2 scans exported per feature ? Additionally, most of the file size is in the MS1 due to high background, and it does not seem require to export the full range of the MS1 scans, can we export only the range around the precursor ion ion (peak value), like m/z 150 or so ?

**Comment by tomas-pluskal** [comment]
I used the SIRIUS GUI (version 3.5.1). Try loading the file I pasted above into SIRIUS and see or yourself.

**Comment by tomas-pluskal** [comment]
Additional comment: wouldn't it be useful to have an option to export a single feature from the peak list? I think the most common use case is for the user to identify a single feature that comes up.

<img width="1241" alt="screen shot 2017-10-09 at 14 50 58" src="https://user-images.githubusercontent.com/1796692/31353649-4e4902f6-ad01-11e7-99c2-32efd86d77b5.png">

**Comment by lfnothias** [comment]
@tomas-pluska, we are going to switch to mzML format instead of .MGF for both the Sirius and GNPS export module. Could you indicate if there is a already an .mzML format export feature in MZmine that we can build on the of it ?
@kaibioinfo @abisarvepalli

**Comment by tomas-pluskal** [comment]
Sure:

- export of whole files: https://github.com/mzmine/mzmine2/blob/master/src/main/java/net/sf/mzmine/modules/rawdatamethods/rawdataexport/RawDataExportTask.java#L80
- export of individual spectrum: https://github.com/mzmine/mzmine2/blob/master/src/main/java/net/sf/mzmine/modules/visualization/spectra/ExportSpectraTask.java#L181

I can guide you if you have any questions.

**Comment by lfnothias** [comment]
Thank you very much. We are on it.

**Comment by tomas-pluskal** [comment]
@lfnothias @abisarvepalli 
Any progress on this? I would like to release MZmine 2.30 relatively soon, because of some important changes in the R connection layer.

**Comment by lfnothias** [comment]
@tomas-pluskal  Right. Please give us couple of days so we finalize and double-check.

**Comment by jeep3** [comment]
Hello, @ttomas-pluskal MZmine 2.30 have an option to export a single feature from the peak list (export for sirius to analyze single compound fragmentation trees reloaded)?

**Comment by tomas-pluskal** [comment]
Not yet. I hope @lfnothias is going to add this option soon :)

**Comment by XsirdanielX** [comment]
@jeep3, till we have this option an easy way to great a single feature sirius input would be to export teh complete peak list to .mgf and copying the desired scan number in a new .mgf (text file). 
An example for feature number 2 of my list:


BEGIN IONS
FEATURE_ID=2
PEPMASS=415.2117656360973
RTINSECONDS=451.5589090909091
SCANS=2
CHARGE=1+
MSLEVEL=1
415.2115936279297 4.87688864E8
416.215087890625 1.37178432E8
417.21771240234375 2.3000778E7
418.2210388183594 3001823.25
END IONS

BEGIN IONS
FEATURE_ID=2
PEPMASS=415.2117656360973
SCANS=2
RTINSECONDS=451.5589090909091
CHARGE=1+
MSLEVEL=2
77.03877258300781 93828.4140625
78.04681396484375 76232.625
79.05442810058594 496195.65625
81.03375244140625 532681.4375
83.04939270019531 506259.65625
85.02865600585938 462602.84375
87.04408264160156 95626.234375
91.05429077148438 5067427.0
93.07008361816406 334246.0
94.63313293457031 41669.3515625
95.04914855957031 257552.609375
97.02845764160156 277806.15625
99.04396057128906 211929.78125
101.05976104736328 49856.6640625
103.05423736572266 693587.0625
104.06218719482422 663634.0625
105.06967163085938 289504.96875
106.05589294433594 50556.11328125
107.08562469482422 4209391.5
109.0651626586914 140271.453125
110.65563201904297 43784.44140625
111.04438018798828 559494.875
115.05390167236328 393774.0625
117.07002258300781 1554106.0
119.08562469482422 1.75082368E8
127.03905487060547 433206.40625
129.0545196533203 572684.625
133.06488037109375 2568055.75
135.08050537109375 7741300.5
145.0500946044922 59071.296875
147.06553649902344 622575.8125
151.07591247558594 98193.546875
157.10174560546875 48463.55859375
173.0963134765625 52515.90625
247.0955047607422 74100.0703125
251.1462860107422 41161.9375
277.1064453125 171016.5
281.1382141113281 531296.6875
295.115478515625 331346.03125
399.5000915527344 45746.14453125
415.2120056152344 3560982.25
END IONS

**Comment by jeep3** [comment]
I got it, thank you @XsirdanielX.
